src/Components/Canvas.js

```diff
@@ -57,14 +58,15 @@ drawUpdatedImage = () =>  {
     })
 
     imageObj.crossOrigin = "anonymous"
-    imageObj.src = this.props.urlValue;
+    imageObj.src = urlValue;
   }
 }
 
 handleChange = (e) => {
   this.setState({urlValue: e.target.value});
 }
 render () {
+  const { clickedColorsArray, deleteItem, bigColor, urlValue, handleChange } = this.props;
   this.drawUpdatedImage();
```

Comment:
this should go in componentDidUpdate, and you should check to see if prevProps.urlValue !== this.props.urlValue.  The fact that you put this in render() is why your image updates immediately on pasting in.  render() gets called each time anything on the page changes at all.